### ISSUE TITLE
pie chart legends moving upwards when chart re-rendered

### ISSUE THREAD

**Issue by asgeo1** [DmitryBaranovskiy/g.raphael#154]
I'm finding if I re-render a pie chart via a setTimeout, that the legends are moving upwards.

I currently have them in the "east" position.

It's this line here:

```
var bb = chart.labels.getBBox()
```

For some reason, the height gets calculated wrong - each new call to the function that creates the pie chart, the bbox.height is twice as large as before.

As a work-around, if I comment out the transform:

```
//chart.labels.translate.apply(chart.labels, tr);
```

then it works OK - i.e. I don't think the transform is needed for the "east" position anyway.

**Comment by vedmack** [comment]
Seems that Raphael  2.1.1 version fixes that somehow... I had the same issue with several pie charts... you can get it from here http://raphaeljs.com/raphael.js (here where I found it https://groups.google.com/forum/?fromgroups#!topic/raphaeljs/5nrN37tRrCA)

**Comment by nambad** [comment]
Basically it is because of the change in values in the original object. You need to make sure that everytime you re-render a chart, the original values are passed to Raphael. The original options object is passed by reference and hence changes the object. I use a clone object technique to solve this issue.

**Comment by aamirafridi** [comment]
https://twitter.com/aamirafridi/status/232855136635084800

**Comment by scottksmith95** [comment]
I agree with vedmack. Version 2.1.1 does fix this issue. For me, the legend was moving to the left when it was positioned south.

**Comment by MaffooBristol** [comment]
:+1: Upgrading did the trick for me too. I was confused and saddened by why my legend would want to run away from me like that :cry:
